packages/components/toggle-group/src/toggle-group.ts

```diff
@@ -0,0 +1,80 @@
+import { type SlToggleEvent } from '@sl-design-system/shared/events.js';
+import { ToggleButton, type ToggleButtonSize } from '@sl-design-system/toggle-button';
+import { type CSSResultGroup, LitElement, type PropertyValues, type TemplateResult, html } from 'lit';
+import { property } from 'lit/decorators.js';
+import styles from './toggle-group.scss.js';
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'sl-toggle-group': ToggleGroup;
+  }
+}
+
+/**
+ * A component for visually grouping buttons together.
+ *
+ * @slot default - The default slot.
+ */
+export class ToggleGroup extends LitElement {
+  /** @internal */
+  static override styles: CSSResultGroup = styles;
+
+  get #buttons(): ToggleButton[] {
+    return (
+      this.renderRoot
+        .querySelector('slot')
+        ?.assignedElements({ flatten: true })
+        .filter((element): element is ToggleButton => element instanceof ToggleButton) ?? []
+    );
+  }
+
+  /** If set, will disable all buttons in the group. */
+  @property({ type: Boolean, reflect: true }) disabled?: boolean;
+
+  /**
+   * By default, only a single toggle button inside the group can be active. This means
+   * that the group will automatically deactivate the other buttons when one is toggled.
+   *
+   * If you toggle this property, then multiple buttons can be active at the same time.
+   * In this case the group does nothing when a button is toggled. Use this mode if you want to
+   * handle the toggling of buttons yourself.
+   */
+  @property({ type: Boolean }) multiple?: boolean;
+
+  /** Determines the size of all buttons in the group. */
+  @property({ reflect: true }) size?: ToggleButtonSize;
+
+  override updated(changes: PropertyValues<this>): void {
+    super.updated(changes);
+
```

Comment:
Maybe `role="group"` needs to be applied?

Reply:
I looked at https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/group_role, but that doesn't really match with the toggle group?

Reply:
`"region"` is apparently the right one: https://lea.verou.me/blog/2022/07/button-group/ && https://twitter.com/LeonieWatson/status/1545788775644667904